### PR TITLE
Fixed third-party Footprint bug (sequential login after user creation) with trailpack-sequelize

### PR DESCRIPTION
**Pull request by LobeTia**

### CODE REVIEW

**Comment by LobeTia** [comment]
I'm having troubles while writing some good tests for this use case, tomorrow I will try to mock Google responses
The PR is ok and can be merged for now, but I prefer to write some coverage for the future 😉

File: lib/passports.js [hunk 1]
```diff
@@ -105,14 +105,20 @@ module.exports = {
...
-              }).catch(next)
+              // I save any update to the Passport before moving on and
+              // I read the associated user instance
+              return Promise.all([
+                app.services.FootprintService.findAssociation("Passport", passport.id, "User"),
+                app.services.FootprintService.update("Passport", passport.id, passport)
+              ])
+                .then(results => {
+                  let userInstance = results[0]; // Not so usefull, just for keeping namings clear
+                  return app.services.FootprintService.find('user', userInstance.id)
```
Comment (jaumard) [inline lib/passports.js]: Why do you do a find again  ? You already have the user under `userInstance ` right ?

**Comment by jaumard** [comment]
If you manage to create tests for third party login/registration if would be awesome ! I didn't find a way to do it and don't have time to dig in it. Thanks for your help/work

**Comment by jaumard** [comment]
I think this issue fix https://github.com/jaumard/trailpack-passport/issues/108

File: lib/passports.js [hunk 2]
```diff
@@ -105,14 +105,20 @@ module.exports = {
...
-                  .then(user => {
-                    next(null, user)
-                  })
-                  .catch(next)
-              }).catch(next)
+              // I save any update to the Passport before moving on and
+              // I read the associated user instance
+              return Promise.all([
+                app.services.FootprintService.findAssociation("Passport", passport.id, "User"),
+                app.services.FootprintService.update("Passport", passport.id, passport)
```
Comment (JAertgeerts) [inline lib/passports.js]: With postgres, this update is not persisted. If you amend passport with passport.get(), it does work.

Comment (jaumard) [inline lib/passports.js]: what is `.get` exactly ? Maybe a `toJSON()` would be more appropriate to be multiple ORM compliant right ?

Comment (JAertgeerts) [inline lib/passports.js]: toJSON works too, just tested

Comment (jaumard) [inline lib/passports.js]: Then go for toJSON :D almost sure `get` will not work with other ORM ^^ thanks for testing !

Comment (LobeTia) [inline lib/passports.js]: `.toJSON` works great and I thinks that remove the needs of the various `passport.tokens = JSON.stringify(query.tokens)` around in the code.
But for now I will change only this row, I'd like to have some good coverage before starting to refactor the trailpack

File: lib/passports.js (hunk 1 above)
Comment (LobeTia) [inline lib/passports.js]: Good sighting, I think that it's redundant, I will remove it in the next commit

**Pull request merged**